# Sales Analysis for Adventure Works

## Importing Library and dataset

In [1]:
import pandas as pd
df = pd.read_csv('Internet Sales.csv')
df.head()

/Users/asadtariq/opt/anaconda3/envs/pythonProject/lib/python3.7/site-packages/IPython/core/interactiveshell.py:3444: DtypeWarning: Columns (14) have mixed types.Specify dtype option on import or set low_memory=False.
  exec(code_obj, self.user_global_ns, self.user_ns)


Unnamed: 0  CustomerKey  GeographyKey         City CountryRegionCode  \
0           0        11000            26  Rockhampton                AU   
1           1        11000            26  Rockhampton                AU   
2           2        11000            26  Rockhampton                AU   
3           3        11000            26  Rockhampton                AU   
4           4        11000            26  Rockhampton                AU   

  EnglishCountryRegionName CustomerAlternateKey Title FirstName MiddleName  \
0                Australia           AW00011000   NaN       Jon          V   
1                Australia           AW00011000   NaN       Jon          V   
2                Australia           AW00011000   NaN       Jon          V   
3                Australia           AW00011000   NaN       Jon          V   
4                Australia           AW00011000   NaN       Jon          V   

   ... ProductStandardCost  TotalProductCost SalesAmount    TaxAmt  Freight  \
0  ...           1912.1544         1912.1544     3399.99  271.9992  84.9998   
1  ...           1265.6195         1265.6195     2319.99  185.5992  57.9998   
2  ...              8.2205            8.2205       21.98    1.7584   0.5495   
3  ...           1481.9379         1481.9379     2384.07  190.7256  59.6018   
4  ...             10.8423           10.8423       28.99    2.3192   0.7248   

  CarrierTrackingNumber CustomerPONumber   OrderDate     DueDate    ShipDate  
0                   NaN              NaN  2011-01-19  2011-01-31  2011-01-26  
1                   NaN              NaN  2013-01-18  2013-01-30  2013-01-25  
2                   NaN              NaN  2013-01-18  2013-01-30  2013-01-25  
3                   NaN              NaN  2013-05-03  2013-05-15  2013-05-10  
4                   NaN              NaN  2013-05-03  2013-05-15  2013-05-10  

[5 rows x 60 columns]

### Checking names of data columns

In [2]:
df.columns

Index(['Unnamed: 0', 'CustomerKey', 'GeographyKey', 'City',
       'CountryRegionCode', 'EnglishCountryRegionName', 'CustomerAlternateKey',
       'Title', 'FirstName', 'MiddleName', 'LastName', 'NameStyle',
       'BirthDate', 'MaritalStatus', 'Suffix', 'Gender', 'EmailAddress',
       'YearlyIncome', 'TotalChildren', 'NumberChildrenAtHome',
       'EnglishEducation', 'SpanishEducation', 'FrenchEducation',
       'EnglishOccupation', 'SpanishOccupation', 'FrenchOccupation',
       'HouseOwnerFlag', 'NumberCarsOwned', 'AddressLine1', 'AddressLine2',
       'Phone', 'DateFirstPurchase', 'CommuteDistance', 'ProductKey',
       'EnglishProductName', 'OrderDateKey', 'DueDateKey', 'ShipDateKey',
       'CustomerKey.1', 'PromotionKey', 'CurrencyKey', 'SalesTerritoryKey',
       'SalesOrderNumber', 'SalesOrderLineNumber', 'RevisionNumber',
       'OrderQuantity', 'UnitPrice', 'ExtendedAmount', 'UnitPriceDiscountPct',
       'DiscountAmount', 'ProductStandardCost', 'TotalProductCost',
       '

### Extracting year from Order Date.

In [3]:
df['OrderDate'] = pd.DatetimeIndex(df['OrderDate']).year

In [5]:
df.head()

Unnamed: 0  CustomerKey  GeographyKey         City CountryRegionCode  \
0           0        11000            26  Rockhampton                AU   
1           1        11000            26  Rockhampton                AU   
2           2        11000            26  Rockhampton                AU   
3           3        11000            26  Rockhampton                AU   
4           4        11000            26  Rockhampton                AU   

  EnglishCountryRegionName CustomerAlternateKey Title FirstName MiddleName  \
0                Australia           AW00011000   NaN       Jon          V   
1                Australia           AW00011000   NaN       Jon          V   
2                Australia           AW00011000   NaN       Jon          V   
3                Australia           AW00011000   NaN       Jon          V   
4                Australia           AW00011000   NaN       Jon          V   

   ... ProductStandardCost  TotalProductCost SalesAmount    TaxAmt  Freight  \
0  ...           1912.1544         1912.1544     3399.99  271.9992  84.9998   
1  ...           1265.6195         1265.6195     2319.99  185.5992  57.9998   
2  ...              8.2205            8.2205       21.98    1.7584   0.5495   
3  ...           1481.9379         1481.9379     2384.07  190.7256  59.6018   
4  ...             10.8423           10.8423       28.99    2.3192   0.7248   

  CarrierTrackingNumber CustomerPONumber  OrderDate     DueDate    ShipDate  
0                   NaN              NaN       2011  2011-01-31  2011-01-26  
1                   NaN              NaN       2013  2013-01-30  2013-01-25  
2                   NaN              NaN       2013  2013-01-30  2013-01-25  
3                   NaN              NaN       2013  2013-05-15  2013-05-10  
4                   NaN              NaN       2013  2013-05-15  2013-05-10  

[5 rows x 60 columns]

-----------------------------------------------------------------------------------------
### 1. What are the internet sales by Year for Each country Region?

In [31]:
result = df.groupby(['CountryRegionCode','OrderDate'])['SalesAmount'].sum().reset_index(name ='Total Sales')
print(result)

   CountryRegionCode  OrderDate   Total Sales
0                 AU       2010  2.090978e+04
1                 AU       2011  2.563732e+06
2                 AU       2012  2.128407e+06
3                 AU       2013  4.339443e+06
4                 AU       2014  8.507720e+03
5                 CA       2010  3.578270e+03
6                 CA       2011  5.715718e+05
7                 CA       2012  3.076045e+05
8                 CA       2013  1.085633e+06
9                 CA       2014  9.457620e+03
10                DE       2011  5.205002e+05
11                DE       2012  6.086580e+05
12                DE       2013  1.761876e+06
13                DE       2014  3.277830e+03
14                FR       2010  3.399990e+03
15                FR       2011  4.108453e+05
16                FR       2012  6.480655e+05
17                FR       2013  1.578512e+06
18                FR       2014  3.195060e+03
19                GB       2010  6.990982e+02
20                GB       2011  5

----------------------------------------------------------------------
### 2. Generate summary statistics for Yearly Income, gender by Country Regions

In [22]:
df.groupby(['CountryRegionCode','Gender'])['YearlyIncome'].describe()

count          mean           std      min  \
CountryRegionCode Gender                                                 
AU                F        6685.0  67060.583396  30613.956169  10000.0   
                  M        6660.0  65412.912913  32550.778171  10000.0   
CA                F        3648.0  58651.315789  19435.158438  10000.0   
                  M        3972.0  61460.221551  22702.851581  10000.0   
DE                F        2820.0  48287.234043  37622.814569  10000.0   
                  M        2805.0  46730.837790  37487.619366  10000.0   
FR                F        2781.0  39367.134124  28421.836245  10000.0   
                  M        2777.0  39805.545553  28665.660622  10000.0   
GB                F        3401.0  56518.670979  51130.689140  10000.0   
                  M        3505.0  55771.754636  50963.707551  10000.0   
US                F       10682.0  64766.897585  25498.805483  10000.0   
                  M       10662.0  65450.196961  26520.348175  10000.0   

                              25%      50%      75%       max  
CountryRegionCode Gender                                       
AU                F       40000.0  70000.0  90000.0  160000.0  
                  M       40000.0  70000.0  90000.0  170000.0  
CA                F       40000.0  60000.0  70000.0  170000.0  
                  M       40000.0  60000.0  80000.0  170000.0  
DE                F       20000.0  30000.0  80000.0  130000.0  
                  M       20000.0  30000.0  50000.0  130000.0  
FR                F       20000.0  30000.0  40000.0  110000.0  
                  M       20000.0  30000.0  40000.0  110000.0  
GB                F       20000.0  30000.0  80000.0  170000.0  
                  M       20000.0  30000.0  80000.0  170000.0  
US                F       50000.0  60000.0  70000.0  170000.0  
                  M       50000.0  60000.0  70000.0  170000.0

----------------------------------------------------------------------
### 3. What is the average price of product category that generates the most sales.

In [42]:
x = df.groupby(['ProductKey'])['SalesAmount'].sum().reset_index(name ='Sales')
x.sort_values(['Sales'], ascending = False)

ProductKey       Sales
10          312  1205876.99
8           310  1202298.72
11          313  1080637.54
12          314  1055589.65
9           311  1005493.87
..          ...         ...
106         530     7425.12
92          480     7307.39
96          484     7218.60
93          481     2679.02
94          482     2427.30

[158 rows x 2 columns]

Maximum sales are of Product 312. Average price of this product is calculate below.

In [38]:
result2 = df.groupby(['ProductKey'])['SalesAmount'].mean().reset_index(name ='Average Sales')
result2= result2.sort_values(['Average Sales'], ascending = False)
result2.head(1)



ProductKey  Average Sales
10         312        3578.27

### Average Sales of Product '312' is 3578.27

----------------------------------------------------------------------
### 4. Does owning a home makes a difference in sales?


checking count of transactions for both House Owners and Non House Owners.

In [43]:
df['HouseOwnerFlag'].value_counts()

1    41699
0    18699
Name: HouseOwnerFlag, dtype: int64

In [44]:
result3 = df.groupby(['HouseOwnerFlag'])['SalesAmount'].sum().reset_index(name ='Sales')
print(result3)

   HouseOwnerFlag         Sales
0               0  8.973327e+06
1               1  2.038535e+07


Owning a home does not make a lot of difference, the total count of home owners is 2.3 times the non home owners, and the sales almost show the same difference. Which indicates that the average sale for both Home owners and Non home owners is almost similar. But the number of Sales by home owners is quite high as compared to non home owners, which indicate that House Owners are more prone to make a sale, hence the revenue generated by Home owners is quite high as compared to Non House Owners.